app/V1Module/presenters/UsersPresenter.php

```diff
@@ -430,4 +432,23 @@ public function actionExercises(string $id) {
     $this->sendSuccessResponse($user->getExercises()->getValues());
   }
 
+  public function checkInvalidateTokens(string $id) {
+    $user = $this->users->findOrThrow($id);
+
+    if (!$this->userAcl->canInvalidateTokens($user)) {
+      throw new ForbiddenRequestException();
+    }
+  }
+
+  /**
+   * Invalidate all existing tokens issued for given user
+   * @POST
+   * @param string $id Identifier of the user
+   */
+  public function actionInvalidateTokens(string $id) {
+    $user = $this->users->findOrThrow($id);
+    $user->setTokenValidityThreshold(new DateTime());
```

Comment:
If this endpoint is used from UI, it also invalidates current user session. It might be beneficial, if the action returns a new (valid) token.